### PR TITLE
Phase2-hgx119 HGCDigi can work with generic DetId

### PR DESCRIPTION
**Pull request by bsunanda**
DetId class changed for HGC from HGCal to HGCSilicon and from Hcal to HGCScintillator. Accordingly Digi collection now depends on generic DetId

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@fabiocos This may be the last change to DataFormats

**Comment by kpedro88** [comment]
@bsunanda I do think this makes more sense in general. But a few questions/comments:
1) Removing the old definitions prevents us from reading old files.
2) If the first template parameter of the `HGCDataFrame` class will always be `DetId`, it doesn't need to be a template parameter at all.

However, if we keep the old dataframe types defined, we can keep the first template parameter in `HGCDataFrame`. Then we could just have a new `typedef HGCDataFrame<DetId,HGCSample> HGCalDataFrame` or something, and use that everywhere in the code from now on.

**Comment by bsunanda** [comment]
@kpedro88 I agree with you. I shall restore the old definitions in classes.h and the xml files. Do I need to change the typedef for DigiCollections? Are Digi's stored as persistent objects?

**Comment by kpedro88** [comment]
Yes, digis are stored persistently, so it's better not to change the typedefs.

**Comment by kpedro88** [comment]
@bsunanda will you update the digitizer code (`HGCDigitizer`, `HGCDigiProducer`, `PreMixingHGCalWorker`) and local reco code (`HGCalUncalibRecHit...` classes) to use the new digi collection type for the new geometry?
See https://github.com/cms-sw/cmssw/commit/f2629c3be15c0ee8eac502e4b88220235175f788 and https://github.com/cms-sw/cmssw/commit/6a053c5b1a50bff507e089b853113efde6159a9a from when I changed the BH collection in #23545.

**Comment by bsunanda** [comment]
@kpedro88 Could you please take care of them? I am concentrating on geometry issue now

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda okay, I will do it once this is merged.

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

File: DataFormats/HGCDigi/src/classes.h [hunk 1]
```diff
@@ -18,22 +24,6 @@ namespace DataFormats_HGCDigi {
     std::vector<HGCDataFrame<HcalDetId,HGCSample> > vHGCalBHDataFrames;
     edm::SortedCollection< HGCDataFrame<HcalDetId,HGCSample> > scHGCalBHDataFrames;
     edm::Wrapper< edm::SortedCollection< HGCDataFrame<HcalDetId,HGCSample> > > prodHGCalBHDataFrames;
-
```
Comment (fabiocos) [inline DataFormats/HGCDigi/src/classes.h]: @bsunanda @kpedro88 were these DataFormats already used in past productions?

Comment (kpedro88) [reply DataFormats/HGCDigi/src/classes.h]: @fabiocos yes, that's why we kept them here.

**Comment by bsunanda** [comment]
I tried to explain that the  removed ones were not used at least for 2 years once the hexagonal design is chosen and became operational. The codes for square cells do not exist in other parts of CMSSW. So there is no scope to use them anywhere anymore
________________________________

**Comment by fabiocos** [comment]
+1

**Pull request merged**